conans/model/version.py

```diff
@@ -128,31 +127,19 @@ def main(self):
 
     @property
     def major(self):
-        try:
-            return self.main[0]
-        except IndexError:
-            return None
+        return self.main[0] if len(self.main) > 0 else None
 
     @property
     def minor(self):
-        try:
-            return self.main[1]
-        except IndexError:
-            return None
+        return self.main[1] if len(self.main) > 1 else None
 
     @property
     def patch(self):
-        try:
-            return self.main[2]
-        except IndexError:
-            return None
+        return self.main[2] if len(self.main) > 2 else None
 
     @property
     def micro(self):
-        try:
-            return self.main[3]
-        except IndexError:
-            return None
+        return self.main[3] if len(self.main) > 3 else None
```

Comment:
same. This could be argued that ``micro`` is not that popular, but in this case, ``micro`` is not even queried that much, and it is better to keep it aligned with the above.